tofhir-server/src/main/scala/io/tofhir/server/service/mapping/ProjectMappingFolderRepository.scala

```diff
@@ -229,10 +234,17 @@ class ProjectMappingFolderRepository(mappingRepositoryFolderPath: String, projec
       files.foreach { file =>
         val source = Source.fromFile(file, StandardCharsets.UTF_8.name()) // read the JSON file
         val fileContent = try source.mkString finally source.close()
-        val fhirMapping = fileContent.parseJson.extract[FhirMapping]
-        // discard if the mapping id and file name not match
-        if (FileOperations.checkFileNameMatchesEntityId(fhirMapping.id, file, "mapping")) {
-          fhirMappingMap.put(fhirMapping.id, fhirMapping)
+        // Try to parse the file content as FhirMapping
+        try {
+          val fhirMapping = fileContent.parseJson.extract[FhirMapping]
+          // discard if the mapping id and file name not match
+          if (FileOperations.checkFileNameMatchesEntityId(fhirMapping.id, file, "mapping")) {
+            fhirMappingMap.put(fhirMapping.id, fhirMapping)
+          }
+        }catch{
+          case _: JsonParseException =>
```

Comment:
How can I reproduce this error ?

I utilize an empty json file and it gives me a org.json4s.MappingException exception. Let's catch all exception with case e:Throwable and log the original error as well because it provides the real problem in the json like "No usable value for id"